File: src/main/java/com/ourcompany/jpa/Document.java
package com.ourcompany.jpa;

import com.ourcompany.jpa.embedded.Owner;
import jakarta.persistence.*;


import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "documents")
public class Document {
    @Column(nullable = false, insertable = false, updatable = false)
    @Id
    public Integer id = 0;

    @Embedded
    public Owner owner = new Owner();

}

File: src/main/kotlin/com/ourcompany/jpa/Document.kt
package com.ourcompany.jpa

import com.ourcompany.jpa.embedded.Owner
import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

@Entity
@Table(name = "documents")
class Document {
    @Column(nullable = false, insertable = false, updatable = false)
    @Id
    var id: Int = 0

    @Embedded
    var owner: Owner = Owner()

    @OneToMany(mappedBy = "document")
    var recipients: MutableList<DocumentReceiver> = mutableListOf()
}
